Client/Frontend/ContentBlocker/ContentBlockerHelper+Whitelist.swift

```diff
@@ -17,56 +46,69 @@ extension ContentBlockerHelper {
             return ""
         }
         // Note that * is added to the front of domains, so foo.com becomes *foo.com
-        let list = "'*" + ContentBlockerHelper.whitelistedDomains.joined(separator: "','*") + "'"
-        return ", {'action': { 'type': 'ignore-previous-rules' }, 'trigger': { 'url-filter': '.*', 'unless-domain': [\(list)] }}".replacingOccurrences(of: "'", with: "\"")
+        let list = "'*" + ContentBlockerHelper.whitelistedDomains.domainSet.joined(separator: "','*") + "'"
+        return ", {'action': { 'type': 'ignore-previous-rules' }, 'trigger': { 'url-filter': '.*', 'if-domain': [\(list)] }}".replacingOccurrences(of: "'", with: "\"")
     }
 
-    func whitelist(enable: Bool, url: URL, completion: (() -> Void)? = nil) {
-        guard let domain = url.baseDomain else { return }
+    static func whitelist(enable: Bool, url: URL, completion: (() -> Void)?) {
+        guard let domain = whitelistableDomain(fromUrl: url) else { return }
+
         if enable {
-            ContentBlockerHelper.whitelistedDomains.insert(domain)
+            whitelistedDomains.add(domain)
         } else {
-            ContentBlockerHelper.whitelistedDomains.remove(domain)
+            whitelistedDomains.delete(domain)
         }
 
         updateWhitelist(completion: completion)
     }
 
-    private func updateWhitelist(completion: (() -> Void)?) {
-        TPStatsBlocklistChecker.shared.updateWhitelistedDomains(Array(ContentBlockerHelper.whitelistedDomains))
-
+    static func clearWhitelist(completion: (() -> Void)?) {
+        whitelistedDomains.clear()
+        updateWhitelist(completion: completion)
+    }
+    
+    private static func updateWhitelist(completion: (() -> Void)?) {
         removeAllRulesInStore {
-            self.compileListsNotInStore {
-                NotificationCenter.default.post(name: .ContentBlockerUpdateNeeded, object: nil)
+            compileListsNotInStore {
                 completion?()
+                NotificationCenter.default.post(name: .ContentBlockerTabSetupRequired, object: nil)
+
             }
         }
 
         guard let fileURL = ContentBlockerHelper.whitelistFileURL() else { return }
-        if ContentBlockerHelper.whitelistedDomains.isEmpty {
+        if ContentBlockerHelper.whitelistedDomains.domainSet.isEmpty {
             try? FileManager.default.removeItem(at: fileURL)
             return
         }
 
-        let list = ContentBlockerHelper.whitelistedDomains.joined(separator: "\n")
+        let list = ContentBlockerHelper.whitelistedDomains.domainSet.joined(separator: "\n")
         do {
             try list.write(to: fileURL, atomically: true, encoding: .utf8)
         } catch {
             Sentry.shared.send(message: "Failed to save whitelist file")
         }
     }
+    // Ensure domains used for whitelisting are standardized by using this function.
+    static func whitelistableDomain(fromUrl url: URL) -> String? {
```

Comment:
Does using `url.baseDomain` not make sense. That is how we fetch a base domain in other parts of the app.

Reply:
`Does using url.baseDomain not make sense.` -> I was asked to match desktop, thus `host`